### PR TITLE
refactor(collapse): add docblocks to component implementation

### PR DESCRIPTION
**Pull request by leandrogrillo**

### CODE REVIEW

File: src/js/components/collapse.js [hunk 1]
```diff
@@ -10,12 +10,23 @@ const DEFAULTS = {
 }
 
 class Collapse {
+  /**
+   * Create Collapse component.
+   *
```
Comment (fabricionaweb) [inline src/js/components/collapse.js]: `Create a new Collapse instance.`

File: src/js/components/collapse.js [hunk 2]
```diff
@@ -52,6 +75,11 @@ class Collapse {
     }
   }
 
+  /**
+   * Set isCollapsed flag to true, set target maxHeight to 0
+   * Remove visible and active class from target and element
+   * emit `collapse:hide` event.
+   */
```
Comment (fabricionaweb) [inline src/js/components/collapse.js]: Esse comentario ta errado, ta duplicado com o de baixo

File: src/js/components/collapse.js [hunk 3]
```diff
@@ -64,6 +92,11 @@ class Collapse {
     emitter.emit('collapse:hide', this.element, this.toggle)
   }
 
+  /**
+   * Set isCollapsed flag to false, set maxHeight to component
```
Comment (fabricionaweb) [inline src/js/components/collapse.js]: Coloca isCollapsed dentro de `

File: src/js/components/collapse.js [hunk 4]
```diff
@@ -10,12 +10,23 @@ const DEFAULTS = {
 }
 
 class Collapse {
+  /**
+   * Create a new Collapse instance.
+   *
+   * @param  {Element} element Element to be used as the toggle component.
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Could it be `collapse` instead of `toggle`?

File: src/js/components/collapse.js [hunk 5]
```diff
@@ -10,12 +10,23 @@ const DEFAULTS = {
...
+   * @param  {Object}  options Options used to customize the component.
+   */
   constructor (element, options) {
     this.element = element
 
     this.options = { ...DEFAULTS, ...options }
   }
 
+  /**
+   * Set initial component state, listeners and target
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Period at the end

File: src/js/components/collapse.js [hunk 6]
```diff
@@ -10,12 +10,23 @@ const DEFAULTS = {
...
+   *
+   * @return {Collapse} Class instance
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Description could be more explanatory like `An instance of the Collapse class.`

File: src/js/components/collapse.js [hunk 7]
```diff
@@ -28,6 +39,12 @@ class Collapse {
     return this
   }
 
+  /**
+   * Check if component is opened or closed by default and set `isCollapsed` flag
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Period at the end.

File: src/js/components/collapse.js [hunk 8]
```diff
@@ -28,6 +39,12 @@ class Collapse {
...
+   * Get target `scrollHeight` and set as `toggleHeight` property
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Period

File: src/js/components/collapse.js [hunk 9]
```diff
@@ -28,6 +39,12 @@ class Collapse {
...
+   * If `isCollapsed` flag is false, set target `maxHeight` based on `toggleHeight`
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Period

File: src/js/components/collapse.js [hunk 10]
```diff
@@ -40,10 +57,16 @@ class Collapse {
     }
   }
 
+  /**
+   * Add @toggleTarget as listener for the events setted in options.
```
Comment (willamesoares) [inline src/js/components/collapse.js]: `set`

File: src/js/components/collapse.js [hunk 11]
```diff
@@ -52,6 +75,11 @@ class Collapse {
     }
   }
 
+  /**
+   * Set `isCollapsed` flag to true, set target `maxHeight` to 0
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Comma at the end

File: src/js/components/collapse.js [hunk 12]
```diff
@@ -52,6 +75,11 @@ class Collapse {
...
+   * remove visible and active class from target and element
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Comma at the end

File: src/js/components/collapse.js [hunk 13]
```diff
@@ -64,6 +92,11 @@ class Collapse {
     emitter.emit('collapse:hide', this.element, this.toggle)
   }
 
+  /**
+   * Set `isCollapsed` flag to false
```
Comment (willamesoares) [inline src/js/components/collapse.js]: Can you use something to connect the sentences? You can separate them with commas or finish each sentence with a period `.` and then start the others with capital letters. Think about generating a documentation page for this component using a tool like JSDoc for instance. If you write sentences in this way you're going to end up with a grammatically incorrect paragraph for that documentation.

**Pull request merged**